### ISSUE TITLE
database/sql: *Rows.Err() does not return errors from *Rows.Scan()

### ISSUE THREAD

**Issue by zombiezen** [golang/go#25787]
I'm not really advocating for the behavior to change, but the usage of the word "iteration" in the current documentation of [`Rows.Err()`](https://golang.org/pkg/database/sql/#Rows.Err) is ambiguous with respect to the behavior of `*Rows.Scan()`.  More examples and/or more precise documentation would help.

### What version of Go are you using (`go version`)?

go version go1.10.2 linux/amd64

### Does this issue reproduce with the latest release?

Yes

### What did you do?

https://play.golang.org/p/1xbP8gEMNtb

### What did you expect to see?

(Given a certain reading of the `Rows.Err()` docs:)

```
2018/06/07 10:24:45 Error from Scan(...): sql: Scan error on column index 0: sql/driver: couldn't convert 42 into type bool
2018/06/07 10:24:45 Error from Err(): sql: Scan error on column index 0: sql/driver: couldn't convert 42 into type bool
```

### What did you see instead?

```
2018/06/07 10:24:45 Error from Scan(...): sql: Scan error on column index 0: sql/driver: couldn't convert 42 into type bool
2018/06/07 10:24:45 No error from Err()
```

**Comment by bcmills** [comment]
CC @bradfitz @kardianos for SQL, @robpike for prose

**Comment by kardianos** [comment]
There is a change on master that may address this. Scan should return errs now if rows.lasterr == nil.
Does this help?

**Comment by kardianos** [comment]
Oh wait, you want the opposite. You would like Scan to store the scan error in lasterr if lasterr is nil so it can get reported in rows.Err().

**Comment by bcmills** [comment]
Yep. It's not obvious that the phrase “during iteration” is intended to exclude `Scan`, since the `Scan` call occurs after the start of iteration and before the end.
